src/dmd/iasmdmd.d

```diff
@@ -178,23 +179,23 @@ version (none) // don't use bReturnax anymore, and will fail anyway if we use re
             // get the first part of an expr
             if (asmstate.tokValue != TOK.endOfFile)
             {
-                asm_cond_exp(opnd1);
-                o1 = &opnd1;
+                asm_cond_exp(opnds[0]);
+                nOps = 1;
                 if (asmstate.tokValue == TOK.comma)
                 {
                     asm_token();
-                    asm_cond_exp(opnd2);
-                    o2 = &opnd2;
+                    asm_cond_exp(opnds[1]);
+                    nOps = 2;
                     if (asmstate.tokValue == TOK.comma)
                     {
                         asm_token();
-                        asm_cond_exp(opnd3);
-                        o3 = &opnd3;
+                        asm_cond_exp(opnds[2]);
+                        nOps = 3;
                         if (asmstate.tokValue == TOK.comma)
                         {
                             asm_token();
-                            asm_cond_exp(opnd4);
-                            o4 = &opnd4;
```

Comment:
surely this can be converted into a loop

Reply:
Ha, of course.  I think the one I came up with is equiavalent